### ISSUE TITLE
Make batchedFetchFromRemoteEndpoint public.

### ISSUE THREAD

**Issue by return007** [apollographql/apollo-client#1806]
Why is that `batchedFetchFromRemoteEndpoint` made private? Please let the developers override their functionality and make changes as per their requirements.

**Intended outcome:**
The method `fetchFromRemoteEndpoint` is public and so I expected this for batching interface too.

**Actual outcome:**
`fetchFromRemoteEndpoint` is private...

**Comment by helfer** [comment]
Hey @return007 can you tell me what you would like to do specifically with `batchedFetchFromRemoteEndpoint`? If we have a specific use-case that is reasonable, we can definitely make it public.

**Comment by return007** [comment]
@helfer I want to override the functionality of method and modify the request body (to some format that is acceptable by my graphql endpoint) before sending it. I also thought of doing this in a Middleware but came to know that its not possible.

**Comment by helfer** [comment]
So what you would want is a way to override the batchedFetchFromRemoteEndpoint in your BatchNetworkInterface to provide your own fetch function? If that's your use-case, I think for now you can probably just copy the whole BatchNetworkInterface, because there's not that much code in it. We're working on a new thing called apollo-fetcher which will make your use-case a lot easier. If you want, it would be great if you can head over there and leave a comment/issue to say that you would like to be able to provide your own fetch function to the batch network interface.

**Comment by return007** [comment]
Hey @helfer, initially I did the same as you said (copying `BatchNetworkInterface` in my own code. But later I tried to override the `batchedFetchFromRemoteEndpoint ` method and it worked (to my surprise). Sorry for the issue. 

And yes, if its the case the you guys are working on apollo-fetcher (that would make this use case easier), I would love to see that. 

I think the use case should be mentioned in the examples. I would love to do that :smile:
